export/cloud_datastore_admin_test.py

```diff
@@ -0,0 +1,145 @@
+
+        Returns:
+            *. A one-time use context manager.
+        """
+        response = requests.Response()
+        response._content = content # pylint: disable=protected-access
+        response.status_code = status_code
+        return self.swap(requests, 'post', lambda *args, **kwargs: response)
+
+    def mock_http_exception(self, error_message):
+        """Return a context manager that mocks out all HTTP responses by raising
+        an exception.
+
+        Args:
+            error_message: str. The content of the exception to raise.
+
+        Returns:
+            *. A one-time use context manager.
+        """
+        def raise_exception(*unused_args, **unused_kwargs):
+            """Always raises an exception."""
+
+            raise Exception(error_message)
+
+        return self.swap(requests, 'post', raise_exception)
+
+    def test_export_as_admin_succeeds(self):
+        admin_ctx = self.login_context(self.ADMIN_EMAIL, is_super_admin=True)
+        prod_ctx = self.mock_production_environment()
+        http_mock = self.mock_http_response('success')
+
+        with admin_ctx, prod_ctx, http_mock:
+            response = (
+                self.testapp.get('/cloud_datastore_export?bucket=gs://abc'))
+        self.assertEqual(response.status_code, 200)
+
+    def test_export_as_cron_job_succeeds(self):
+        cron_job_headers = {b'X-AppEngine-Cron': b'true'}
+        prod_ctx = self.mock_production_environment()
+        http_mock = self.mock_http_response('success')
+
+        with prod_ctx, http_mock:
+            response = self.testapp.get(
+                '/cloud_datastore_export?bucket=gs://abc',
+                headers=cron_job_headers)
+        self.assertEqual(response.status_code, 200)
+
+    def test_export_when_http_request_raises_error_fails(self):
+        admin_ctx = self.login_context(self.ADMIN_EMAIL, is_super_admin=True)
+        prod_ctx = self.mock_production_environment()
+        http_mock = self.mock_http_exception('uh-oh!')
+
+        with admin_ctx, prod_ctx, http_mock:
+            response = self.testapp.get(
+                '/cloud_datastore_export?bucket=gs://abc',
+                expect_errors=True)
+        self.assertEqual(response.status_code, 500)
+
+    def test_export_outside_of_prod_environment_fails(self):
+        admin_ctx = self.login_context(self.ADMIN_EMAIL, is_super_admin=True)
+
+        with admin_ctx:
+            response = self.testapp.get(
+                '/cloud_datastore_export?bucket=gs://abc', expect_errors=True)
+        self.assertEqual(response.status_code, 404)
+
+    def test_export_with_no_bucket_fails(self):
+        admin_ctx = self.login_context(self.ADMIN_EMAIL, is_super_admin=True)
+        prod_ctx = self.mock_production_environment()
+
+        with admin_ctx, prod_ctx:
+            response = self.testapp.get(
+                '/cloud_datastore_export', expect_errors=True)
+        self.assertEqual(response.status_code, 400)
```

Comment:
For 400s please assert the error message too

Reply:
Done.